sqlalchemy_redshift/dialect.py

```diff
@@ -900,7 +901,16 @@ def create_connect_args(self, *args, **kwargs):
 class RedshiftDialect_psycopg2(
     Psycopg2RedshiftDialectMixin, PGDialect_psycopg2
 ):
-    pass
+    @classmethod
+    def dbapi(cls):
```

Comment:
Can this be defined in `Psycopg2RedshiftDialectMixin` or does it have to be a classmethod on each concrete dialect? I assume this is some hook in the SQLAlchemy machinery, but I'm struggling to find documentation about a dbapi classmethod. Can you point to background on this approach?

Reply:
I think we could move this to `Psycopg2RedshiftDialectMixin` if we did something like this, using the value of `driver` defined by the dialect (e.g. defined [here](https://github.com/sqlalchemy/sqlalchemy/blob/fb81f9c8d914f9911925dd3f4e77d7fc374b267c/lib/sqlalchemy/dialects/postgresql/psycopg2cffi.py#L29) for the psycopg2cfifi dialect).

```python
class Psycopg2RedshiftDialectMixin(RedshiftDialectMixin): 
   @classmethod
    def dbapi(cls):
        try:
            return __import__(self.driver)
        except ImportError:
            raise ImportError(
                'No module named {}.'.format(self.driver)
            )
```

For the `dbapi` classmethod, looking to the `DefaultDialect` class (from which [`PGDialect`](https://github.com/sqlalchemy/sqlalchemy/blob/fb81f9c8d914f9911925dd3f4e77d7fc374b267c/lib/sqlalchemy/dialects/postgresql/base.py#L3076) inherits from) it is used in [`DefaultDialect.connect()`](https://github.com/sqlalchemy/sqlalchemy/blob/bf1fe670513abeb1596bc5266f50db1ffe62f3bd/lib/sqlalchemy/engine/default.py#L582), as well as for defining the [dbapi paramstyle that will be used](https://github.com/sqlalchemy/sqlalchemy/blob/bf1fe670513abeb1596bc5266f50db1ffe62f3bd/lib/sqlalchemy/engine/default.py#L307).

I was unable to find any docs for the `dpapi` classmethod either, as I stumbled upon it when looking for where the driver's `connect` method is called.